api/src/config/playwright.js

```diff
@@ -2,10 +2,11 @@
  * This module stores the configuration for using Puppeteer to scrape Google Scholar.
  */
 
-const puppeteerConfig = {
+const playwrightConfig = {
   noOfDummyLinks: 4, // first 4 links aren't publications can be ignored
   noOfThreads: 10, // number of concurrent threads
```

Comment:
Can the number of threads be set more dynamically than a literal value? Do you know if there's a way to find the optimal thread count given hardware limitations?
In a C/C++ multirhreaded app, you could call `std::thread::hardware_concurrency()` to find the max number of threads that can run in parallel. Anything more will degrade performance due to constant thread switching

Reply:
Actually, I don't think I'm using noOfThreads anymore and forgot to remove it. Right now it's using promises to scrape in 'parallel' in line 175 of controllers/publications.js

Reply:
Playwright as far as I saw doesn't have anything in built to handle parallelism so I used the solution mentioned here: https://github.com/microsoft/playwright/issues/4729